lib/reader/src/parser.rs

```diff
@@ -903,6 +931,68 @@ impl<'a> Parser<'a> {
         // Collect comments for the next instruction to be allocated.
         self.gather_comments(ctx.function.dfg.next_inst());
 
+        let mut encoding: Option<Encoding> = None;
+        let mut result_registers: Option<Vec<&'a str>> = None;
+
+        // encoding ::= "[" (encoding_literal | "-") , (result_registers | "-") "]"
```

Comment:
This is the right grammar. Allow either to be `"-"` independently. We can always verify this more carefully in the verifier.